### ISSUE TITLE
Bug Report: Error trying to deploy a service

### ISSUE THREAD

**Issue by franciscocpg** [serverless/serverless#4599]
# This is a Bug Report

## Description

* What went wrong?
After upgrading to `1.25.0` I'm getting the error `An error occurred: StatsDashcollectorLambdaVersionNsKKgFybFLzbcHve8GZNamV5yX9o2NShqs7sUdqPs - A version for this Lambda function exists ( 5 ). Modify the function to create a new version.` when running `sls deploy` to deploy a service that already exists. Rolling back to `1.24.1` and everything works as expected. Just a guess, but is it related to https://github.com/serverless/serverless/pull/4510?

* What did you expect should have happened?
The service should deploy.

* What was the config you used?

* What stacktrace or error message from your provider did you see?
An error occurred: StatsDashcollectorLambdaVersionNsKKgFybFLzbcHve8GZNamV5yX9o2NShqs7sUdqPs - A version for this Lambda function exists ( 5 ). Modify the function to create a new version.

Similar or dependent issues:

## Additional Data

* ***Serverless Framework Version you're using***:
1.25.0

* ***Operating System***:
linux 

* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by HyperBrain** [comment]
@franciscocpg Thanks for reporting.

#4510 only includes the function configuration additionally into the hash generation for the version resource. Before it did not change if you only had new function configs, but only when you changes the sources.

Did you modify only a function config, and that led to the failure?
In 1.24.1 changing the function config never was deployed to a new version (which was the bug), so all versions with only config changes were not persisted at all and thus were invalid.

Can you try the following:
(1) change a function's sources => deploy
(2) change a function's sources and config => deploy
(3) change only a function's config (e.g. memory) => deploy

**Comment by HyperBrain** [comment]
wait .... do you use the alias plugin?

**Comment by franciscocpg** [comment]
Hi @HyperBrain 
I'm just using `serverless-webpack` plugin.

Anyway, I've tried to deploy now and it worked. 
Also I've tried the 3 scenarios you mentioned:
(1) change a function's sources => deploy
(2) change a function's sources and config => deploy
(3) change only a function's config (e.g. memory) => deploy
and it worked too.

I'm unable to reproduce the problem again. So I'm going to close this one.

**Comment by HyperBrain** [comment]
@franciscocpg Thanks for checking again.

If you encounter the problem again, we can reopen the issue. I wonder if it is something with the creation order in CF, that can be wrong in very rare occasions. Maybe there is still a hidden bug with missing CF dependencies in Serverless, especially with the function, version and role resources.

**Comment by dashmug** [comment]
I have the same problem. The only thing I changed was the `serverless` and `serverless-webpack` versions.

I have not yet investigated deeply whether this is caused by `serverless` or `serverless-webpack`.

This is my diff.

```
-    "serverless": "1.25.0",
-    "serverless-webpack": "^4.2.0",
+    "serverless": "1.24.1",
+    "serverless-webpack": "^4.1.0",
```

I haven't changed any function source (maybe `webpack` did) nor function config.

**Comment by HyperBrain** [comment]
@dashmug As I wrote above, it could be a general missing DependsOn in the resources created by Serverless. Did it work for you on a second deploy, or if you change the sources and redeploy?
Additionally, please check the "Events" tab in the CF stack console and see, if it correctly removes the old version resource, _before_ creating the new one. That should show unambiguously that it is some dependency issue, that is in already for a longer time, but now is reveiled by the correct version handling.
If so, we have to analyze where a DependsOn has to be added by Serverless.

**Comment by coulonxyz** [comment]
Hi, 
I was able to reproduce this issue. I have a function with a reservedConcurrency set to 2 and when I change it to 1, I have the following error:
An error occurred: SomeLambdanWpvYF5EVWNUeiNcxCbg5T842oRMvkpjhfou1BCAW8 - A version for this Lambda function exists ( 75 ). Modify the function to create a new version..
I was able to workaround this issue by removing the function from my serverless.yml file, deploy, and then re-add and redeploy.

That was ok as it was in dev but it would have been quite a bummer I i had to do it in production.

Julien
